### PR TITLE
Mitigate specific LSF test flakyness

### PR DESCRIPTION
**Pull request by berland**
On local hardware the test is more aggressive to allow faster test times, implying some flakyness when local hardware is bogged down in tasks. That residual flakyness we can solve better with reruns instead of increasing the wait times in the test.

**Issue**
Resolves  locally observed flakyness.

**Approach**
Rerun ♻️ 

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: tests/integration_tests/scheduler/test_lsf_driver.py [hunk 1]
```diff
@@ -257,6 +257,7 @@ def mock_poll_once_by_bhist(*args, **kwargs):
     assert job_id in driver._bhist_cache
 
 
+@pytest.mark.flaky(reruns=5)
```
[the remark this answers is not in the document]
Comment (berland) [reply tests/integration_tests/scheduler/test_lsf_driver.py:260]: We could add the marker programmatically in `conftest.py` where we check for test function name and avoid `darwin` and `--lsf`. Not sure that extra test code noise is worth it.

[the remark this answers is not in the document]
Comment (sondreso) [reply tests/integration_tests/scheduler/test_lsf_driver.py:260]: Agreed, just a bit worried that we might introduce flakyness in the actual code en then the tests will not catch it :thinking:

File: tests/integration_tests/scheduler/test_lsf_driver.py [hunk 2]
```diff
@@ -271,6 +272,7 @@ async def test_kill_before_submit_is_finished(
         job_kill_window = 5
         test_grace_time = 10
     else:
+        # More aggressive on local iron, fixing residual flakyness with reruns
```
Comment (xjules) [inline tests/integration_tests/scheduler/test_lsf_driver.py:275]: As discussed let's try to replace `await asyncio.sleep(0.01)` with the actual lock being set; ie. `self._submit_locks[0]`

Comment (berland) [reply tests/integration_tests/scheduler/test_lsf_driver.py:275]: Turns out it that is not enough to avoid flakyness

**Comment by berland** [comment]
The flakyness is easily reproducible by adding  `@pytest.mark.repeat(100)` to the offending test, and then running: `pytest tests/integration_tests/ -sxv -n 12 `.

Awaiting the submit locks does not help, as there are more sources of flakiness. E.g. if the execution of the kill itself is too slow, the first process will survive no matter how.  Can't even remove flakyness by increasing job_kill_window and test_grace_time.

**Comment by berland** [comment]
Added a new proposal which is not pretty, but works.

File: tests/integration_tests/scheduler/test_lsf_driver.py [hunk 3]
```diff
@@ -271,8 +281,8 @@ async def test_kill_before_submit_is_finished(
         job_kill_window = 5
         test_grace_time = 10
     else:
-        job_kill_window = 1
-        test_grace_time = 2
+        job_kill_window = 2
```
Comment (xjules) [inline tests/integration_tests/scheduler/test_lsf_driver.py:284]: That is not really a fix :smile:

**Review by xjules — APPROVED** [review]
Alight! Let's see how long this will last :rocket:

**Pull request merged**